JCP: Install Jetpack intro screen

itsmeichigo opened the pull request:
Part of #5365 

# Description
This PR adds the initial screen of the Install Jetpack flow. This screen is currently presented when tapping on Install Jetpack action button on the Jetpack Benefits screen.

@adamzelinski - can you please check out if it's okay to keep the accent colors for buttons on dark mode? On the design both "Cancel" and "Get Started" buttons are white, but those do not match button styles on the app.

# Screenshots
| Light mode | Dark mode |
| ----- | ----- |
| <img src="https://user-images.githubusercontent.com/5533851/142553275-ef956c30-a74a-4a29-822c-f044ee353d98.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/142553314-2bbd37f6-9aea-4bf9-a28a-215de6e90641.png" width=320 />

# Testing
1. Make sure that `.jetpackConnectionPackageSupport` feature flag is on.
2. Tap on Jetpack Benefits banner, which will redirect you to Jetpack Benefits screen.
3. Tap on Install Jetpack button on Jetpack Benefits screen, notice that you are redirected to a new screen for introduction to Install Jetpack flow.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Add JetpackInstallIntroView

itsmeichigo committed:
Add attributed text for description of Install Jetpack intro

itsmeichigo committed:
Add hosting controller for JetpackInstallIntroView

itsmeichigo committed:
Navigate from Jetpack benefits to Jetpack install

itsmeichigo committed:
Fix typo for JetpackInstallHostingController

itsmeichigo committed:
Fix issue with Attributed text color and font

itsmeichigo committed:
Update padding for action button

ghost commented:
@itsmeichigo yes thank you, can we please use our current dark mode buttons for the dark design (pink30) Here is the design. Could you also please check the text styling, logo, and padding as they don't seem to match the designs from the screenshot you've posted. Cheers

<img width="1187" alt="Screen Shot 2021-11-19 at 1 52 18 pm" src="https://user-images.githubusercontent.com/74938219/142556971-ddf06e22-39d9-4409-99df-1b4d50572123.png">

jaclync commented on WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/AttributedText.swift:
```diff
@@ -132,10 +132,10 @@ private struct TextViewWrapper: UIViewRepresentable {
         uiView.maxLayoutWidth = maxLayoutWidth
         uiView.font = context.environment.font?.uiFont ?? UIFont.preferredFont(forTextStyle: .body)
         uiView.textColor = context.environment.foregroundColor.map(UIColor.init)
+        uiView.attributedText = attributedText
```
nit: any reasons for this move?

itsmeichigo replied:
From my testing, the font and text color set in the attributed text is overriden when setting font and text color on the label afterwards. Moving this makes sure to avoid that while keeping the font and text color set to the label in cases when no font or text color is set to the attributed text.

jaclync commented on WooCommerce/Classes/ViewRelated/Dashboard/JetpackInstall/JetpackInstallIntroView.swift:
```diff
@@ -0,0 +1,116 @@
+
+    enum Localization {
+        static let closeButton = NSLocalizedString("Close", comment: "Title of the Close action on the Jetpack Install view")
+        static let installAction = NSLocalizedString("Get Started", comment: "Title of install action in the Jetpack benefits view.")
```
nit: Jetpack Install view?
```suggestion
        static let installAction = NSLocalizedString("Get Started", comment: "Title of install action in the Jetpack Install view.")
```

jaclync commented on WooCommerce/Classes/ViewRelated/Dashboard/JetpackInstall/JetpackInstallIntroView.swift:
```diff
@@ -0,0 +1,116 @@
+        static let installAction = NSLocalizedString("Get Started", comment: "Title of install action in the Jetpack benefits view.")
+        static let installTitle = NSLocalizedString("Install Jetpack", comment: "Title of the Install Jetpack intro view")
+        static let installDescription = NSLocalizedString("Install the free Jetpack plugin to %1$@ and experience the best mobile experience.",
+                                                          comment: "Description of the Jetpack Install flow for the specified site")
```
nit: maybe we can mention what goes into the placeholder?

```suggestion
                                                          comment: "Description of the Jetpack Install flow for the specified site. The %1$@ is the site address.")
```

jaclync commented on WooCommerce/Classes/ViewRelated/Dashboard/JetpackInstall/JetpackInstallIntroView.swift:
```diff
@@ -0,0 +1,116 @@
+                    .resizable()
+                    .renderingMode(.template)
+                    .foregroundColor(Color(.jetpackGreen))
+                    .frame(width: Constants.jetpackLogoSize, height: Constants.jetpackLogoSize)
```
nit: maybe we can let the logo auto-adjust its size with system font size via `@ScaledMetric`

jaclync commented on WooCommerce/Classes/ViewRelated/Dashboard/JetpackInstall/JetpackInstallIntroView.swift:
```diff
@@ -0,0 +1,116 @@
+    // Closure invoked when Close button is tapped
+    var dismissAction: () -> Void = {}
+
+    let siteURL: String
```
nit: maybe this can be in the initializer so that the property isn't accessible externally?

itsmeichigo committed:
Add green jetpack icon

itsmeichigo committed:
Update jetpack logo image

jaclync commented on WooCommerce/Classes/ViewRelated/Dashboard/JetpackInstall/JetpackInstallIntroView.swift:
```diff
@@ -0,0 +1,116 @@
+        )
+        let boldSiteAddress = NSAttributedString(string: siteName, attributes: [.font: boldFont, .foregroundColor: UIColor.text])
+        attributedString.replaceFirstOccurrence(of: siteName, with: boldSiteAddress)
+        return attributedString
```
can't wait to be able to use the iOS 15+ markdown `Text` feature!

itsmeichigo replied:
Yes me too! But with markdown we can't really specify different colors for different parts of the text so we'd have to compromise the fancy design for the sake of simplicity 😅

jaclync reviewed (approved):
LGTM, the screen looks great! 🚢

itsmeichigo committed:
Make title bold and add extra spacing between the logo and title

itsmeichigo committed:
Make logo image scalable

itsmeichigo committed:
Add extra padding for cancel button

itsmeichigo committed:
Add initializer for JetpackInstallIntroView

itsmeichigo committed:
Update Jetpack logo on Jetpack Benefits banner

itsmeichigo commented:
Thank you @adamzelinski  and @jaclync for the reviews! I have added new changes based on the comments, including addition of the green Jetpack logo. I've also updated the Jetpack Benefits banner to use this logo instead the old one with transparent lightning inside.

| Benefits banner | Install screen |
| ----- | ----- |
| <img src="https://user-images.githubusercontent.com/5533851/142562687-7187258e-310f-47ee-99bf-33bdce1a2c38.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/142562857-c376ed60-daf2-4a20-89b6-8189bfb7bafa.png" width=320 /> |
